### DISCUSSION TITLE
[06/01] 피어세션 일지

### DISCUSSION THREAD

**Discussion by newspring97** [bcaitech1/p4-dkt-lastffang#54] [Peer Session]
## **브리핑**

- **[@ddoyoon](https://github.com/ddoyoon)**: 강의 따라잡기. time delta 부분 구현 중 → 한 번 실험해볼 때마다 시간이 오래 걸림. Time delta branch에 푼 시간이 20분 이상 걸린 문제들은 값을 0으로 바꾸어 다시 값을 적절한 값으로 채워넣으려고 하는 중. (연속형 값 → Linear & LayerNorm)
- **[@scarlettJin](https://github.com/scarlettJin)**: Feature 추가 노력 (Time 관련) but 도윤님이 겪고 있는 issue와 비슷한 문제 발생 → 성능이 잘 나오지 않는 현상 마주함. 긴 시간동안 문제를 푸는 경우가 생각보다 많음 → 고민. 강의 듣기.
- **[@alinghi](https://github.com/alinghi)**: 수업 듣기. Riiid 2등 solution 포팅 시도 중.
- **[@gonipark](https://github.com/gonipark)**: 근형님 코드 리뷰. `is_train=False` 인 경우, 모든 answerCode의 값이 unknown으로 들어가는 문제 → 확인 필요 (dataframe 안에 answerCode가 unknown으로 바뀌는 문제가 궁금)
- **[@newspring97](https://github.com/newspring97)**: Continuous Embedding 텐서플로우 → 파이토치로 코드 구현 중
- **[@GY-Jeong](https://github.com/GY-Jeong)**: 연속형 범주 feature 추가하는 코드 구현. 버그 fix & 진홍님의 방식과 차이로 동일 성능이 나오지 않아 코드 다시 살펴보는 중.

## Office Hour - Q&A

- 내일로 밀림

## Inference.py의 결과 문제

- 기존의 baseline code는 inference 시에 여러번 실행해도 결과값이 바뀌지 X
- 근영님 PR의 경우 `setSeed` 없이 inference 실행 시 매번 결과가 달라지는 현상이 발생
    - Inference의 경우 `nn.Embedding` 에서 랜덤 때문에 엉뚱한 값을 주는 문제
    - Inference는 저장된 값을 불러오는 건데 왜? → embedding의 경우 값을 저장하는게 아니라 `setSeed` 만 해주면 됨.
    - 근데 왜 이전에는 멀쩡하게 작동했지??? (random seed 문제는 없던 것으로 추정됨)

## Continuous Feature를 넣어도 성능향상이 미비한 문제

- 근영님 PR의 경우 기존 진홍님 방식에 비해 성능이 0.02 정도 낮은 상황
- 진홍님 방식을 적용하여 실험해보기로

## Continuous Feature를 제외했을 때 성능이 더 좋은 문제

- Continuous Feature가 없는 경우의 성능이 넣었을 때보다 좋은 현상 발견

## 새로운 모델 시도해보기

- BERT + LSTM 모델

## `is_train=False` 인 경우, 모든 answerCode의 값이 unknown으로 들어가는 문제

- `__preprocessing` 에서 `df[col] = df[col].apply(lambda x: x if x in le.classes_ else 'unknown')` 라인을 `df[col] = df[col].apply(lambda x: x if str(x) in le.classes_ else 'unknown')` 으로 바꾸는 경우 knowledgeTag가 모두 `unknown` 으로 나옴
    - 근형님 branch 코드
    - 왜 string을 했는가? → string으로 치환하지 않는 경우 unknown이 값이 unknown으로 설정됨. (이 문제가 KnowledgeTag에서도 발생하여 KnowledgeTag 정보가 사라지는 문제 발생)
    - 진홍님이 baseline에서 코드 수정 후 결과 공유

**Comment by GY-Jeong** [comment]
정리왕이시네요,,, 👍 감사합니당
